Client/Frontend/AuthenticationManager/PasscodeEntryViewController.swift

```diff
@@ -16,6 +16,8 @@ import SwiftKeychainWrapper
 /// Presented to the to user when asking for their passcode to validate entry into a part of the app.
 class PasscodeEntryViewController: BasePasscodeViewController {
     weak var delegate: PasscodeEntryDelegate?
+    var success: (() -> Void)?
+    var cancel: (() -> Void)?
```

Comment:
Makes sense. I think that we should stick with one approach though - either callback blocks or a delegate since at the moment they provide the same functionality. I can be convinced of either option. I think one of the benefits of using the delegate is that it's easier to extend with additional callback methods if we need to reflect different states besides success/cancel and can be easier to prevent retain cycles when passing in closures. Downside is you'll need to extend the option setting classes with the delegate. Mind you it might be better to encapsulate that logic inside the setting classes instead of having it inline in the `passcodeEnabledSettings()` method.